sample/composeApp/src/commonMain/kotlin/com/mifos/passcode/sample/chooseAuthOption/ChooseAuthOptionScreenViewmodel.kt

```diff
@@ -0,0 +1,54 @@
+package com.mifos.passcode.sample.chooseAuthOption
+
+import androidx.lifecycle.ViewModel
+import androidx.lifecycle.viewModelScope
+import auth.deviceAuth.RegistrationResult
+import com.mifos.passcode.auth.deviceAuth.PlatformAuthenticationProvider
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.channels.Channel
+import kotlinx.coroutines.flow.receiveAsFlow
+import kotlinx.coroutines.launch
+
+class ChooseAuthOptionScreenViewmodel(
+    private val chooseAuthOptionRepository: ChooseAuthOptionRepository,
+) : ViewModel() {
+
+    private val _registrationResult = Channel<RegistrationResult?>()
+    val registrationResult = _registrationResult.receiveAsFlow()
+
```

Comment:
Is there any reason you use Channel in my point of view u can use `MutableStateFlow` and you can easilty assing it and collect it.

```
 private val _registrationResult = MutableStateFlow<RegistrationResult?>(null)
    val registrationResult: StateFlow<RegistrationResult?> = _registrationResult
```
```
_registrationResult.value = provider.registerUser(
                userId,
                userEmail,
                displayName
            )
```